src/provider/provider.ts

```diff
@@ -97,15 +97,13 @@ export interface Provider {
    * The transaction needs to be signed beforehand.
    * Returns the transaction hash
    * @param {string} tx the base64-encoded signed transaction
-   * @param {TransactionEndpoint} endpoint the transaction broadcast type (sync / commit).
+   * @param {BroadcastType} endpoint the transaction broadcast type (sync / commit).
    * Defaults to broadcast_sync
    */
-  sendTransaction(
+  sendTransaction<BroadcastTransactionResult>(
```

Comment:
I was trying to avoid the situation where the caller would need to cast the response into a specific response type they want (ex `BroadcastTxSyncResult`). This is what I want to achieve:

```ts
const response: BroadcastTxSyncResult = await sendTransaction("...", ...)
```

The user knows which response they want when the call the method, so this syntax allows me to do:

```ts
const response: BroadcastTxSyncResult = await sendTransaction<BroadcastTxSyncResult>("...", ...)
```
in order to explicitly state the return type instead of casting into it.

I'm 99% sure there is a better solution syntax wise. What do you think? How can we make this as seamless as possible for the caller?

Reply:
I'm not sure, but I can think of two ways to do it.

The first is to implement it strictly by overloading another commented function.
The second is to separate the functions. 
I think the reason this happened is that it's hard to infer the response value from the parameters. 

I'll give it some more thought.

Reply:
Resolving this conversation as it's been discussed in the other thread:

https://github.com/gnolang/tm2-js-client/pull/78#discussion_r1319703806